cmake/ccache.cmake

```diff
@@ -11,8 +11,9 @@ if(NOT WIN32)
     # show statistics summary of ccache
     message("ccache version\t\t\t    " ${ccache_version} "\n"
             ${cache_directory})
-    set_property(GLOBAL PROPERTY RULE_LAUNCH_COMPILE ${CCACHE_PATH})
-    set_property(GLOBAL PROPERTY RULE_LAUNCH_LINK ${CCACHE_PATH})
+    set(CMAKE_C_COMPILER_LAUNCHER ${CCACHE_PATH})
+    set(CMAKE_CXX_COMPILER_LAUNCHER ${CCACHE_PATH})
+    set(CMAKE_CUDA_COMPILER_LAUNCHER ${CCACHE_PATH})
   endif()
```

Comment:
Does this speed up the compilation time of flashatten both locally and on CI？

Reply:
For local and NVIDIA's CI, yes. For Paddle's CI, @tianshuo78520a, does Paddle's CI save the ccache results and reuse them? If so, could you help check if this can also reduce Paddle's CI? Thanks.

Reply:
No changes were found in the CI log, but it does not affect PR integration.